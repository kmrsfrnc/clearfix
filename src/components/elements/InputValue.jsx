import React from 'react';
import PropTypes from 'prop-types';
import { TiClipboard } from 'react-icons/ti';

import Clipboardable from './Clipboardable';

import './InputValue.scss';

class InputValue extends React.Component {
  static propTypes = {
    children: PropTypes.string,
  };

  static defaultProps = {
    children: '',
  };

  render() {
    const { children } = this.props;

    return (
      <div className="input-value">
        <input type="text" readOnly value={children} />
        <Clipboardable value={children}>
          <TiClipboard />
        </Clipboardable>
      </div>
    );
  }
}

export default InputValue;
