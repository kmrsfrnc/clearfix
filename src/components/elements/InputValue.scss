.input-value {
  margin-bottom: 0.5em;

  input {
    border-radius: 4px 0 0 4px;
    border-right: 0;
  }

  button {
    border-radius: 0 4px 4px 0;
  }
}