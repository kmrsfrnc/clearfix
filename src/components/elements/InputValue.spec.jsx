import React from 'react';
import { shallow } from 'enzyme';

import InputValue from './InputValue';

describe('InputValue', () => {
  it('matches snapshot', () => {
    expect(shallow(<InputValue>test</InputValue>)).toMatchSnapshot();
  });
});
