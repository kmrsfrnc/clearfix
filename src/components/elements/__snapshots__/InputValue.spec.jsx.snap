// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputValue matches snapshot 1`] = `
<div
  className="input-value"
>
  <input
    readOnly={true}
    type="text"
    value="test"
  />
  <Clipboardable
    value="test"
  >
    <TiClipboard />
  </Clipboardable>
</div>
`;
